knit/src/Knit/Argument.hs

```diff
@@ -238,40 +247,42 @@ lookupArgSingle name args = do
 
 lookupArgOpt ::
        Name
-    -> [Arg (Value components)]
-    -> (Maybe (Value components), [Arg (Value components)])
+    -> [Arg ext (Value components)]
+    -> (Maybe (Value components), [Arg ext (Value components)])
 lookupArgOpt name = \case
     [] -> (Nothing, [])
-    ArgPos a : args -> (Just a, args)
-    arg@(ArgKw name' a) : args ->
+    ArgPos _ a : args -> (Just a, args)
+    arg@(ArgKw _ name' a) : args ->
         if name == name'
         then (Just a, args)
         else over _2 (arg:) $ lookupArgOpt name args
+    XArg _ : args -> lookupArgOpt name args
```

Comment:
I see this handling of `XArg` in `lookupArgOpt` and `lookupArgMany`, but I'm not sure if it's a good idea.

The output of these functions should contain all unprocessed arguments, but you simply discard `XArg` without appending it to the list of processed arguments. If you want to simply ignore them, the correct treatment is this:

```
arg@(XArg _) : args -> over _2 (arg:) $ lookupArgOpt name args
```

However, I'm not sure that ignoring `XArg` is a good idea either. We probably would need to add a class for handling them. However, since it isn't needed for autocompletion, I propose one of two options:

1. Add a `XXArg ext ~ Void` constraint here and call `absurd`.
2. Remove the `XArg` constructor altogether.